dark/views/tournament/info.py

```diff
@@ -12,9 +15,24 @@ class TournamentInfoView(DetailView):
     slug_url_kwarg = 'tournament'
     slug_field = 'id'
 
-    def get_context_data(self, **kwargs):
-        context = super().get_context_data(**kwargs)
-        context['rounds'] = TournamentRound.objects.filter(tournament=self.object)
-        context['contestants'] = Team.objects.filter(tournament=self.object)
-        context['is_creator_viewing'] = self.object.creator == self.request.user
-        return context
+    def get(self, request, *args, **kwargs):
+        if not request.user.is_authenticated:
```

Comment:
LoginRequiredMixin (redirect_to = reverse_lazy('tournament:all') - niekoniecznie tak ale generalnie ten mixin to robi)

Reply:
Zmodyfikowałem ten widok trochę w inny sposób. Mianowicie zawsze info o turnejach było dostępne dla niezalogowanych użytkowników, a ja to usunęłem nie wiadomo po co. Więc wróciłem dostęp do niezalogowanych użytkowników.

Według mnie to nadal nie jest piękny widok. Fajnie byłoby tutaj zastosować coś podobnego do UserAuthenticationDependentContextMixin który byłby kompatybilny z redirect'em. Może wiesz jakiś Mixin który spełnia moje wymagania?

Reply:
Edit: Udało się to zrobić